theano/tensor/basic.py

```diff
@@ -5943,25 +5981,77 @@ def infer_shape(self, node, shapes):
 
 
 def diagonal(a, offset=0, axis1=0, axis2=1):
```

Comment:
the interface of diagonal() function was like this before this PR, are we sure we need to change it? We will finally allow non-default settings for these Ops in the end, we may need to change the interface back then.